packages/frontend/src/components/common/FieldSelect/index.tsx

```diff
@@ -93,8 +101,23 @@ const FieldSelect = <T extends Item = Item>({
     inactiveItemIds = [],
     hasGrouping = false,
     baseTable,
+    focusOnRender = false,
     ...rest
 }: FieldSelectProps<T>): JSX.Element => {
+    const inputRef = useRef<HTMLInputElement | null>(null); // Input ref for focus handling
+
+    // Focus on input when mounted
+    const [isMounted, setIsMounted] = useState(false);
```

Comment:
`isMounted` state is unnecessary. React’s useEffect already runs after the component mounts, so you can directly focus the input inside a useEffect without the additional state.

Reply:
Hmm removing the isMounted state doesn't seem to trigger the focus anymore. but throwing it to the bottom of the event loop with this works. How do you feel about this?

```
 useEffect(() => {
        if (focusOnRender) {
            const interval = setInterval(() => {
                if (inputRef.current) {
                    inputRef.current.focus();
                    clearInterval(interval);
                }
            }, 10);
        }
    }, [focusOnRender]);```

Reply:
@andrewdw 

`setInterval` can be replaced with `setTimeout` and there is no need for `clearInterval`

but also, can you just try this instead? I tested locally and it worked just fine

```ref={(ref) => ref?.focus()}```